dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSReadWriteTest.java

```diff
@@ -126,6 +126,77 @@ public void integrationReadWriteBackTest() throws Exception
         }
     }
 
+    @Test
+    public void readResumeTest() throws Exception
+    {
+        HPCCFile file = new HPCCFile(datasets[0], connString , hpccUser, hpccPass);
+
+        DataPartition[] fileParts = file.getFileParts();
+        if (fileParts == null || fileParts.length == 0)
+        {
+            Assert.fail("No file parts found");
+        }
+
+        FieldDef originalRD = file.getRecordDefinition();
+        if (originalRD == null || originalRD.getNumDefs() == 0)
+        {
+            Assert.fail("Invalid or null record definition");
+        }
+
+        ArrayList<HpccRemoteFileReader.FileReadResumeInfo> resumeInfo = new ArrayList<HpccRemoteFileReader.FileReadResumeInfo>();
```

Comment:
outside the scope of this issue, but should we consider providing a resume info container class with the filepart id and the array of FileReadResumeInfo?

Reply:
Good question, in an actual use case this wouldn't be required as there would be a one-to-one mapping between the FileReadResumeInfo and the file part. We need it here due to the test restarting the read after each record.